MarkupGatewayPlugin.inc.php

```diff
@@ -257,25 +257,98 @@ function fetch($args, $request) {
+			$issueDetails = array (
+						'issue-year'   		=> $issue->getYear(),
+						'issue-volume'  	=> $issue->getVolume(),
+						'issue-number'  	=> $issue->getNumber(),
+						'issue-title'  		=> $issue->getLocalizedTitle(),
+					);
+		}
+		
+		/* Page numbers */
+		$matches = null;
+		if (PKPString::regexp_match_get('/^[Pp][Pp]?[.]?[ ]?(\d+)$/', $submission->getPages(), $matches)) {
+			$matchedPage = htmlspecialchars(Core::cleanVar($matches[1]));
+			$fpage = $matchedPage;
+			$lpage = $matchedPage;
+			$pageCount = 1;
+		} elseif (PKPString::regexp_match_get('/^[Pp][Pp]?[.]?[ ]?(\d+)[ ]?(-|–)[ ]?([Pp][Pp]?[.]?[ ]?)?(\d+)$/', $submission->getPages(), $matches)) {
+			$fpage = htmlspecialchars(Core::cleanVar($matches[1]));
+			$lpage = htmlspecialchars(Core::cleanVar($matches[4]));
+			$pageCount = $fpage - $lpage + 1;
+		}		
+		
+		/* Localized journal titles */
+		foreach ($journal->getName(null) as $loc => $title) {
+			$journalTitles[strtoupper(substr($loc, 0, 2))] = htmlspecialchars(Core::cleanVar($title));
+		}
+		
+		/* Localized article titles */
+		foreach ($submission->getTitle(null) as $loc => $title) {
+			$articleTitles[strtoupper(substr($loc, 0, 2))] = htmlspecialchars(Core::cleanVar($title));
+		}
+		
+		/* Localized abstracts */
+		if (is_array($submission->getAbstract(null))) foreach ($submission->getAbstract(null) as $loc => $abstract) {
+			$abstract = htmlspecialchars(Core::cleanVar(strip_tags($abstract)));
+			if (empty($abstract)) continue;
+			$abstracts[strtoupper(substr($loc, 0, 2))] = $abstract;			
+		}
+		
+		/* TODO: keywords and other classifications */
+		
+		
 		return array (
-				'article-title'     => $submission->getTitle($locale),
-				'abstract'          => $submission->getAbstract($locale),
-				'journal-title'     => $journal->getName($locale),
+				'locale'     		=> $locale,
+				'article-titles'    => $articleTitles,
+				'abstracts'         => $abstracts,
+				'journal-titles'    => $journalTitles,
+				'journal-id'     	=> htmlspecialchars($journal->getSetting('abbreviation', $locale) ? Core::cleanVar($journal->getSetting('abbreviation', $locale)) : Core::cleanVar($journal->getSetting('acronym', $locale))),
 				'institution'       => $journal->getSetting('publisherInstitution'),
 				'contributors'      => $authors,
-				'ISSN'              => $journal->getSetting('onlineIssn'),
-				'journal-id'        => $journal->getId(),
+				'issue-details'     => $issueDetails,
+				'online-ISSN'       => $journal->getSetting('onlineIssn'),
+				'print-ISSN'        => $journal->getSetting('printIssn'),
+				'doi'        		=> $submission->getStoredPubId('doi'),
+				'article-id'        => $submission->getBestArticleId(),
+				'copyright-year'    => $submission->getCopyrightYear(),
+				'copyright-statement'  => htmlspecialchars(__('submission.copyrightStatement', array('copyrightYear' => $submission->getCopyrightYear(), 'copyrightHolder' => $submission->getLocalizedCopyrightHolder(),
```

Comment:
hahah, could be I'll check and test this on my own server this week. Did not have the time to do that in Montreal.